#!/usr/bin/env python
PACKAGE = "vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Params for pickup task
gen.add("yellowLoH", int_t, 0, "Low H bound for box", 47, 0, 255)
gen.add("yellowHiH", int_t, 0, "High H bound for box", 75, 0, 255)
gen.add("yellowLoS", int_t, 0, "Low S bound for box", 70, 0, 255)
gen.add("yellowHiS", int_t, 0, "High S bound for box", 255, 0, 255)
gen.add("yellowLoV", int_t, 0, "Low V bound for box", 120, 0, 255)
gen.add("yellowHiV", int_t, 0, "High V bound for box", 255, 0, 255)

gen.add("yellowLoH2", int_t, 0, "Low H bound for site", 47, 0, 255)
gen.add("yellowHiH2", int_t, 0, "High H bound for site", 75, 0, 255)
gen.add("yellowLoS2", int_t, 0, "Low S bound for site", 70, 0, 255)
gen.add("yellowHiS2", int_t, 0, "High S bound for site", 255, 0, 255)
gen.add("yellowLoV2", int_t, 0, "Low V bound for site", 100, 0, 255)
gen.add("yellowHiV2", int_t, 0, "High V bound for site", 255, 0, 255)

gen.add("greenLoH", int_t, 0, "Low H bound for pickup", 85, 0, 255)
gen.add("greenHiH", int_t, 0, "High H bound for pickup", 145, 0, 255)
gen.add("greenLoS", int_t, 0, "Low S bound for pickup", 120, 0, 255)
gen.add("greenHiS", int_t, 0, "High S bound for pickup", 255, 0, 255)
gen.add("greenLoV", int_t, 0, "Low V bound for pickup", 0, 0, 255)
gen.add("greenHiV", int_t, 0, "High V bound for pickup", 75, 0, 255)

gen.add("redLoH1", int_t, 0, "Low H bound for pickup", 1, 0, 255)
gen.add("redHiH1", int_t, 0, "High H bound for pickup", 25, 0, 255)
gen.add("redLoH2", int_t, 0, "Low H bound for pickup", 100, 0, 255)
gen.add("redHiH2", int_t, 0, "High H bound for pickup", 180, 0, 255)

gen.add("redLoS1", int_t, 0, "Low S bound for pickup", 3, 0, 255)
gen.add("redHiS1", int_t, 0, "High S bound for pickup", 255, 0, 255)
gen.add("redLoV1", int_t, 0, "Low V bound for pickup", 0, 0, 255)
gen.add("redHiV1", int_t, 0, "High V bound for pickup", 255, 0, 255)

gen.add("minSiteArea", int_t, 0, "Minimum site for pickup", 2000, 1000, 300000)
gen.add("minArea", int_t, 0, "Minimum area for samples", 700, 100, 300000)
gen.add("grabbingArea", int_t, 0, "Minimum area for pickup", 10000, 1000, 300000)
gen.add("maxArea", int_t, 0, "Maximum area for samples", 70000, 1000, 300000)

exit(gen.generate(PACKAGE, "pickup", "pickup"))
